scalactic-macro/src/main/scala/org/scalactic/source/PositionMacro.scala

```diff
@@ -1,65 +1,75 @@
-/*
- * Copyright 2001-2016 Artima, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 package org.scalactic.source
 
-import org.scalactic.Resources
 
+//import scala.reflect.macros.blackbox.Context
 import scala.reflect.macros.Context
+import org.scalactic.Resources
 
 /**
  * Helper class for Position macro. (Will be removed from the public API if possible in a subsequent 3.0.0-RCx release.)
  */
-object PositionMacro {
+object PositionMacro {   
+  private object sourceCompatHack {
+    object internal { object decorators {} }
+  }
+  import sourceCompatHack._
+
+  private class PositionMacroImpl(val universe: scala.reflect.api.Universe) {
+
+    private val PositionModule = universe.rootMirror.staticModule("org.scalactic.source.Position")
+    private val PositionClass = universe.rootMirror.staticClass("org.scalactic.source.Position")
+    private val Position_apply = PositionModule.typeSignature.declaration(universe.newTermName("apply"))
+
+    private[scalactic] lazy val showScalacticFillFilePathnames: Boolean = {
+      val value = System.getenv("SCALACTIC_FILL_FILE_PATHNAMES")
+      value != null && value == "yes"
+    }
+
+    def apply(context: Context): context.Tree = {
+      import context.universe._
+
+      // On 2.10, this imports the empty object defined above
+      // On 2.11+, this imports `Tree.setType` extension method
+      import internal.decorators._ 
 
-  private[scalactic] lazy val showScalacticFillFilePathnames: Boolean = {
-    val value = System.getenv("SCALACTIC_FILL_FILE_PATHNAMES")
-    value != null && value == "yes"
+      // Our types don't capture the fact that context.universe eq this.universe, so we have to cast.
+      implicit def castSymbol(s: universe.Symbol): context.universe.Symbol = s.asInstanceOf[context.universe.Symbol]
+      implicit def castType(t: universe.Type): context.universe.Type = t.asInstanceOf[context.universe.Type]
+
+      import treeBuild.{mkAttributedIdent, mkAttributedSelect}
+
+      // Because this macro is called so frequently, we use assign types to the trees as we build them, rather
```

Comment:
s/use assign/assign/

Reply:
@retronym Sorry I don't get this, do you mind to clarify?

Reply:
Change the comment to " // Because this macro is called so frequently, we use assign types as we build them, rather"

Reply:
Updated comment: 

https://github.com/scalatest/scalatest/pull/1690/commits/5b338190018c665d7db3c1f185af7e0bcc818b15